earth_enterprise/src/fusion/autoingest/AssetFactory.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2019 The Open GEE Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef ASSET_FACTORY_H
+#define ASSET_FACTORY_H
+
+#include <common/SharedString.h>
+#include <common/khMetaData.h>
+
+// The goal of this class is to have a single place in the code where all
+// asset creation is handled.
+template<class MutableDerivedAssetHandleType>
+class AssetFactory
```

Comment:
Thinking out loud: since these are static functions, I'm not sure that there's value in wrapping them in a class. It might be fine to make them regular old non-member functions. On the other hand, having the class or a namespace might be a worthwhile organizational tool, especially since the functions have simple names like `Find` and `Make`. I'm undecided.

Reply:
That crossed my mind. I leaned slightly toward consolidating into a class, but maybe a namespace would make more sense.